src/components/map/map-component.tsx

```diff
@@ -260,55 +262,89 @@ export class MapComponent extends BaseComponent<IProps, IState>{
             <TileLayer
                 url="https://services.arcgisonline.com/ArcGIS/rest/services/World_Topo_Map/MapServer/tile/{z}/{y}/{x}"
             />
-            <MapTephraThicknessLayer
-              ref={this.tephraRef}
-              corner1Bound={corner1}
-              corner2Bound={corner2}
-              viewportBounds={viewportBounds}
-              volcanoPos={volcanoPos}
-              gridSize={1}
-              map={this.state.mapLeafletRef}
-              windSpeed={windSpeed}
-              windDirection={windDirection}
-              colHeight={colHeight}
-              mass={mass}
-              hasErupted={hasErupted}
-            />
-          </Pane>
-          isTephraUnit &&
-          <Pane
-            style={{zIndex: 3}}>
-            <Marker
-              position={[volcanoLat, volcanoLng]}
-              icon={iconVolcano}>
-              <Popup>
-                Popup for any custom information.
-              </Popup>
-            </Marker>
-            {cityItems}
-            {pinItems}
-            {riskItems}
-            {sampleLocations}
-            {isSelectingLatlng && <LatLngDrawLayer
-              ref={this.latlngRef}
-              map={this.state.mapLeafletRef}
-              p1Lat={latLngPoint1Lat}
-              p2Lat={latLngPoint2Lat}
-              p1Lng={latLngPoint1Lng}
-              p2Lng={latLngPoint2Lng}
-            /> }
-            {isSelectingCrossSection && <CrossSectionDrawLayer
-              ref={this.crossRef}
-              map={this.state.mapLeafletRef}
-              p1Lat={crossPoint1Lat}
-              p2Lat={crossPoint2Lat}
-              p1Lng={crossPoint1Lng}
-              p2Lng={crossPoint2Lng}
-            /> }
-            {isSelectingRuler && <RulerDrawLayer
-              map={this.state.mapLeafletRef}
-            />}
           </Pane>
+          {
+            isTephraUnit &&
```

Comment:
Would it make sense to separate this into a `TephraMapComponent` and a `SeismicMapComponent` (or whatever the right name would be)?

Reply:
Possibly. I don't want to optimize that quite yet (the tephra unit adds two panes, and the seismic unit one pane respectively, so it's not a lot of code), and there's the possibility that it won't quite make sense, with different parts needing to interweave at different z-indices. Probably each of the panes could be its own component, though, which would simplify the view somewhat.